editor/gui/scene_tree_editor.cpp

```diff
@@ -174,15 +174,42 @@ void SceneTreeEditor::_cell_button_pressed(Object *p_item, int p_column, int p_i
 		EditorDockManager::get_singleton()->focus_dock(NodeDock::get_singleton());
 		NodeDock::get_singleton()->show_groups();
 	} else if (p_id == BUTTON_UNIQUE) {
-		undo_redo->create_action(TTR("Disable Scene Unique Name"));
-		undo_redo->add_do_method(n, "set_unique_name_in_owner", false);
-		undo_redo->add_undo_method(n, "set_unique_name_in_owner", true);
-		undo_redo->add_do_method(this, "_update_tree");
-		undo_redo->add_undo_method(this, "_update_tree");
-		undo_redo->commit_action();
+		bool ask_before_revoking_unique_name = EDITOR_GET("docks/scene_tree/ask_before_revoking_unique_name").operator bool();
```

Comment:
More concise form of casting:

```suggestion
		bool ask_before_revoking_unique_name = bool(EDITOR_GET("docks/scene_tree/ask_before_revoking_unique_name"));
```

Reply:
You don't even need to cast here
```C++
bool ask_before_revoking_unique_name = EDITOR_GET("docks/scene_tree/ask_before_revoking_unique_name");
```